subcommands/devices/cmd.go

```diff
@@ -60,10 +62,28 @@ func NewCommand() *cobra.Command {
 
 	cmd.AddCommand(configCmd)
 	cmd.AddCommand(updatesCmd)
+
+	addUuidFlagToChildren(cmd)
+
 	return cmd
 }
 
-func getDeviceApi(_ *cobra.Command, name string) client.DeviceApi {
+func addUuidFlagToChildren(c *cobra.Command) {
+	ignores := []string{"list-denied", "list", "delete-denied"}
+	for _, child := range c.Commands() {
+		if child.HasSubCommands() {
+			addUuidFlagToChildren(child)
+		} else if !slices.Contains(ignores, child.Name()) {
+			child.Flags().BoolP("by-uuid", "u", false, "Look up device by UUID rather than name")
+		}
+	}
+}
+
+func getDeviceApi(cmd *cobra.Command, name string) client.DeviceApi {
+	byUuid, _ := cmd.Flags().GetBool("by-uuid")
+	if byUuid {
```

Comment:
This code makes an assumption that one of these conditions is met:
- The `by-uuid` flag is present in a command.
- The command without `by-uuid` flag does not call `getDeviceApi`.
- The `byUuid` is always `false` if the `by-uuid` flag is not defined in a command.

Of these, only the third condition is correct, and that only because of the `GetBool` internal implemention detail.
That nuance is IMHO to subtle to rely on it.
I think a more refined code would check for an error (an absence of flag definition) here, like below:
```suggestion
	byUuid, ok := cmd.Flags().GetBool("by-uuid")
	if ok && byUuid {
```

Reply:
maybe we should `panic` if !ok - this would be a programming error, right?

Reply:
If you want to enforce that option - then maybe yes.
IMHO if it were us who see this panic, I'd say yes... but it will the the user who sees it.
I doubt that we want to punish the user because of our own lack of attention.

So, I'd rather assume that command without `by-uuid` only supports lookup by name.
Your code does that already, and I'm fine with that. I just wanted to make it explicit.

Reply:
I force pushed a change to log an error but continue. Here's what it would look like (i forced the `show` command to fail for demo purposes):
```
$ ./bin/fioctl-darwin-arm64 devices show og
ERROR: flag accessed but not defined: by-uuid
UUID:		bfc97632-d493-4f7e-95b0-ec32ef190893
```